test/type-test.ts

```diff
@@ -0,0 +1,91 @@
+import { createTypesFactory } from "../src/index";
```

Comment:
```suggestion
import { createTypesFactory } from "../dist/index";
```
Changing this line breaks the tests.

When changing this line the type of the first argument passed to the `resolveType` changes from `A | B` to `A`. I currently do not have any idea why.